app/test/privacy_policy/table_of_contents/toc_section_expansion_test.dart

```diff
@@ -0,0 +1,593 @@
+              subsections: const [
+                _Section('Xyzzy'),
+              ],
+            )
+          ];
+
+          tocController.build(sections);
+          tocController.markAsCurrentlyRead('Foo');
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Foo', isExpanded: true),
+              _SectionResult('Quz', isExpanded: false),
+            ],
+          );
+        });
+      });
+
+      test(
+          'A section in which a subsection is currently read is expanded automatically.',
+          () {
+        forEveryExpansionBehavior(() {
+          final sections = [
+            _Section(
+              'Foo',
+              subsections: const [
+                _Section('Bar'),
+                _Section('Baz'),
+              ],
+            ),
+            _Section(
+              'Quz',
+              subsections: const [
+                _Section('Xyzzy'),
+              ],
+            )
+          ];
+
+          tocController.build(sections);
+          tocController.markAsCurrentlyRead('Baz');
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Foo', isExpanded: true),
+              _SectionResult('Quz', isExpanded: false),
+            ],
+          );
+        });
+      });
+
+      test(
+          'When manually toggling the expansion state on a collapsed section (currently not read) it expands',
+          () {
+        forEveryExpansionBehavior(() {
+          final sections = [
+            _Section(
+              'Foo',
+              subsections: const [
+                _Section('Bar'),
+                _Section('Baz'),
+              ],
+            ),
+          ];
+          tocController.build(sections);
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Foo', isExpanded: false),
+            ],
+          );
+
+          tocController.toggleExpansionOfSection('Foo');
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Foo', isExpanded: true),
+            ],
+          );
+        });
+      });
+
+      test(
+          'When manually toggling the expansion state on a manually expanded section (currently not read) it collapses',
+          () {
+        forEveryExpansionBehavior(() {
+          final sections = [
+            _Section(
+              'Quz',
+              subsections: const [
+                _Section('Xyzzy'),
+              ],
+            )
+          ];
+          tocController.build(sections);
+
+          tocController.toggleExpansionOfSection('Quz');
+          tocController.toggleExpansionOfSection('Quz');
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Quz', isExpanded: false),
+            ],
+          );
+        });
+      });
```

Comment:
```suggestion
      });

```

Add new line after test.